render/heuristic/heuristic.go

```diff
@@ -0,0 +1,87 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package heuristic
+
+import (
+	"unicode"
+)
+
+// RuneWidth returns the width of a given rune.
+//
+// Non-graphic runes have width of zero. Most graphical runes have the width of
+// one.  Some graphics runes have width of two.  Those include hiragana,
+// katakana, han/kanji and hangul.
+func RuneWidth(r rune) int {
+	if !unicode.IsGraphic(r) {
+		return 0
+	}
+	if !unicode.In(r, unicode.Hiragana, unicode.Katakana, unicode.Hangul, unicode.Han) {
```

Comment:
random comment as I just was looking at this file to see what this PR was about...

This would be easier to read with the doc-comment you have if you inverted the condition like:

```suggestion
	if unicode.In(r, unicode.Hiragana, unicode.Katakana, unicode.Hangul, unicode.Han) {
		return 2
	}
	return 1
```